src/facade/dragonfly_connection.cc

```diff
@@ -784,7 +795,18 @@ void Connection::DispatchFiber(util::FiberSocketBase* peer) {
       }
     };
 
-    builder->SetBatchMode(dispatch_q_.size() > 1);
+    // We really want to have batching in the builder if possible. This is especially
+    // critical in situations where Nagle's algorithm can introduce unwanted high
+    // latencies. However we can only batch if we're sure that there are more commands
+    // on the way. To know if there are, we sometimes yield before executing the last
+    // command in the queue and let the producer fiber push more commands if it wants
+    // to.
+    // As a small optimization, if the queue was never larger than 1, skip the Yield() call.
+    if (dispatch_q_.size() == 1 && queue_was_larger_than_1) {
```

Comment:
can it be that we wake up on non-empty queue and it does not have size > 1 ?
I thought we only push to dispatch_q if we have more than 1 request.

Reply:
At least all pubsub/monitor messages go over the dispatch queue and there we can wake for a single element